### PR TITLE
network: several cleanups for conf parser

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/network/networkd-address-label.c [hunk 1]
```diff
@@ -404,40 +388,67 @@ int config_parse_address_label(
...
-        n->label = k;
-        TAKE_PTR(n);
+        *label = k;
+        return 1;
+}
+
+static const ConfigSectionParser address_label_conf_parser_table[_IPV6_ADDRESS_LABEL_CONF_PARSER_MAX] = {
+        [IPV6_ADDRESS_LABEL]        = { .parser = config_parse_ipv6_address_label,        .ltype = 0, .offset = offsetof(AddressLabel, label), },
+        [IPV6_ADDRESS_LABEL_PREFIX] = { .parser = config_parse_ipv6_address_label_prefix, .ltype = 0, .offset = 0,                             },
+};
```
Comment (poettering) [inline src/network/networkd-address-label.c]: maybe make this a local static var inside of config_parse_ipv6_address_label_section()? you don#t need it outside?

could give it a short name than, and communicate that this is a more local concept

File: src/network/networkd-address.c [hunk 2]
```diff
@@ -2467,7 +2378,18 @@ int config_parse_duplicate_address_detection(
...
+        [ADDRESS_PREFIX_ROUTE]             = { .parser = config_parse_uint32_flag,        .ltype = IFA_F_NOPREFIXROUTE,      .offset = offsetof(Address, flags),                       },
+        [ADDRESS_ADD_PREFIX_ROUTE]         = { .parser = config_parse_uint32_invert_flag, .ltype = IFA_F_NOPREFIXROUTE,      .offset = offsetof(Address, flags),                       },
+        [ADDRESS_AUTO_JOIN]                = { .parser = config_parse_uint32_flag,        .ltype = IFA_F_MCAUTOJOIN,         .offset = offsetof(Address, flags),                       },
+        [ADDRESS_ROUTE_METRIC]             = { .parser = config_parse_uint32,             .ltype = 0,                        .offset = offsetof(Address, route_metric),                },
+        [ADDRESS_NET_LABEL]                = { .parser = config_parse_string,             .ltype = CONFIG_PARSE_STRING_SAFE, .offset = offsetof(Address, netlabel),                    },
+        [ADDRESS_NFT_SET]                  = { .parser = config_parse_nft_set,            .ltype = NFT_SET_PARSE_NETWORK,    .offset = offsetof(Address, nft_set_context),             },
+};
```
Comment (poettering) [inline src/network/networkd-address.c]: here to, maybe make this a local thing?

File: src/shared/conf-parser.c [hunk 3]
```diff
@@ -793,6 +793,32 @@ bool stats_by_path_equal(Hashmap *a, Hashmap *b) {
...
+                int ltype,
+                const char *rvalue,
+                void *userdata) {
+
+        assert(parsers);
+        assert(n_parsers > 0);
+        assert(ltype >= 0);
+        assert((size_t) ltype < n_parsers);
+        assert(userdata);
+
```
Comment (poettering) [inline src/shared/conf-parser.c:814]: might be good having a comment here explaining the situation. i.e. that this is a helper to use for cases where each [] configuration section shall introduce an additional object to synthesize

**Comment by yuwata** [comment]
@poettering Thank you for the review. Addressed all suggestions. Upgrading the green label.

**Comment by yuwata** [comment]
CentOS CI failure is #30573, unrelated.

**Pull request merged**